images/miq-app/ose-assets/appliance-initialize.sh

```diff
@@ -2,14 +2,47 @@
 
 [[ -s /etc/default/evm ]] && source /etc/default/evm
 
-# Environment supplied by OpenShift MIQ via template
-# Assemble DB service host variable based on template parameter $DATABASE_SERVICE_NAME
+# Ensure OpenShift scripting environment is present, exit if source fails
+[[ -s ${SCRIPTS_ROOT}/ose-deploy-common.sh ]] && source ${SCRIPTS_ROOT}/ose-deploy-common.sh || { echo "Failed to source ${SCRIPTS_ROOT}/ose-deploy-common.sh" ; exit 1; }
 
-DATABASE_SVC_HOST="$(echo $DATABASE_SERVICE_NAME | tr '[:lower:]' '[:upper:]')_SERVICE_HOST"
-MEMCACHED_SVC_HOST="$(echo $MEMCACHED_SERVICE_NAME | tr '[:lower:]' '[:upper:]')_SERVICE_HOST"
+# Prepare initialization environment
+prepare_init_env
 
-# Replace memcache host in EVM configuration
-sed -i.bak -E "s/:memcache_server:.*/:memcache_server: ${!MEMCACHED_SVC_HOST}:11211/gi" ${APP_ROOT}/config/settings.yml
+# Check deployment status
+check_deployment_status
 
-echo "Initializing Appliance, please wait ..."
-appliance_console_cli --region ${DATABASE_REGION} --hostname ${!DATABASE_SVC_HOST} --username ${POSTGRESQL_USER} --password ${POSTGRESQL_PASSWORD}
+if [ -n "${UPGRADE}" ]; then
+  echo "== Starting Upgrade =="
+  restore_pv_data
+  setup_memcached
+  migrate_db
+fi
+
+if [ -n "${REDEPLOY}" ]; then
+  echo "== Starting Re-deployment =="
+  restore_pv_data
+  setup_memcached
+  migrate_db
+fi
+
+if [ -n "${NEW_DEPLOYMENT}" ]; then
```

Comment:
DEPLOYMENT_STATUS sounds good, I can actually use a bash "case" instead of multiple if statements, check the value of DEPLOYMENT_STATUS status and proceed accordingly, a default error message can be printed if for any reason DEPLOYMENT_STATUS does not match any (should never happen but JIC).